client/state/themes/actions.js

```diff
@@ -765,3 +768,42 @@ export function confirmDelete( themeId, siteId ) {
 		);
 	};
 }
+
+function setPreviewingThemeData( themeId, source ) {
+	return {
+		type: THEME_PREVIEW_DATA,
+		themeId,
+		source
+	};
+}
+
+export function setPreviewOptions( primary, secondary ) {
+	return {
+		type: THEME_PREVIEW_OPTIONS,
+		primary,
+		secondary
+	};
+}
+
+function showThemePreview() {
+	return {
+		type: THEME_PREVIEW_STATE,
+		isVisible: true
+	};
+}
+
+export function hideThemePreview() {
+	return {
+		type: THEME_PREVIEW_STATE,
+		isVisible: false
+	};
+}
+
+export function preview( themeId ) {
+	return ( dispatch, getState ) => {
+		const isWpcomTheme = !! getTheme( getState(), 'wpcom', themeId );
+		const wpcomOrWporg = isWpcomTheme ? 'wpcom' : 'wporg';
```

Comment:
I think we can move this piece of logic into `ThemePreview`'s `connect` -- we're reading `state` there anyway, so having a dedicated reducer (`themePreviewInfo`) duplicates information inside the Redux tree.